src/main/webapp/app/components/timeSeries/TimeSeries.tsx

```diff
@@ -0,0 +1,186 @@
+            <TimeSeriesEvent
+              key={`${day}-${eventData.createdAt.toString()}`}
+              title={eventData.title}
+              subtitle={eventData.subtitle}
+              createdAt={eventData.createdAt}
+              editBy={eventData.editBy}
+              operation={eventData.operation}
+              bubbleColor={eventData.bubbleColor}
+              content={eventData.content}
+            />
+          ))}
+        </TimeSeriesInfo>
+      ))}
+    </>
+  ) : (
+    <>
+      {days.map(day =>
+        dataForEachDay[day].map((eventData: RequiredTimeSeriesEventData) => (
+          <TimeSeriesInfo key={`${day}-${eventData.createdAt.toString()}`}>
+            <TimeSeriesEvent
+              createdAt={eventData.createdAt}
+              editBy={eventData.editBy}
+              operation={eventData.operation}
+              bubbleColor={eventData.bubbleColor}
+              content={eventData.content}
+              groupByDay={false}
+            />
+          </TimeSeriesInfo>
+        ))
+      )}
+    </>
+  );
+};
+
+export interface ITimeSeriesInfoProps {
+  date?: Date;
+  children: React.ReactNode;
+}
+
+const TimeSeriesInfo = ({ date, children }: ITimeSeriesInfoProps) => {
+  const timelineRef = useRef<HTMLDivElement>(null);
+  const lastChildRef = useRef<HTMLDivElement>(null);
+
+  const [lineHeight, setLineHeight] = useState(0);
+
+  useEffect(() => {
+    function updateLineHeight() {
+      setLineHeight(timelineRef.current.clientHeight - lastChildRef.current.clientHeight - 10); // 10 is height of margin from react-event-timeline
+    }
+
+    updateLineHeight();
+    window.addEventListener('resize', updateLineHeight);
+
+    return () => {
+      window.removeEventListener('resize', updateLineHeight);
+    };
+  }, []);
+
+  return (
+    <Col>
+      {date && (
+        <Row className="pl-1">
+          <h5>
+            <TextFormat value={date} type="date" format={APP_HISTORY_FORMAT} />
+          </h5>
+        </Row>
+      )}
+      <Row>
+        <div ref={timelineRef}>
+          <Timeline
+            style={{ padding: '0px', width: '100%', fontSize: '100%', fontWeight: 'normal' }}
+            lineStyle={{ height: `${lineHeight}px`, top: '10px', left: '10px' }}
+            lineColor="#B0B0B0"
+          >
+            {React.Children.map(children, (child, index) => {
+              if (index === React.Children.count(children) - 1) {
+                return <div ref={lastChildRef}>{child}</div>;
+              }
+              return child;
+            })}
+          </Timeline>
+        </div>
+      </Row>
+    </Col>
+  );
+};
+
+export interface ITimeSeriesEventProps extends Partial<ExtraTimeSeriesEventData> {
+  groupByDay?: boolean;
+}
+
+const TimeSeriesEvent = ({
+  title,
+  subtitle,
+  createdAt,
+  editBy,
+  operation,
+  bubbleColor,
+  content,
+  groupByDay = true,
+}: ITimeSeriesEventProps) => {
+  return (
+    <TimelineEvent
+      style={{ width: '100%' }}
```

Comment:
Do we need to specify to make it work 100%?

Reply:
You're right, I do not. Will update